Make the checkins_interval transient datasets more uniquely named in BigQuery test runs

jeff303 opened the pull request:
Append a timestamp suffix (initialized when the namespace was loaded) to the `checkins_interval_` database names

jeff303 committed:
Make the checkins_interval transient datasets more uniquely named

Append a timestamp suffix (initialized when the namespace was loaded) to the checkins_interval_ database names

pawit-metabase reviewed (changes requested):
Left some comments. I'm not too familiar with the code though, so let me know if I misunderstood something.

jeff303 committed:
Add verbose comments explaining what we're doing

jeff303 committed:
Add transient-dataset? functions for both BQ drivers to put logic in one place

pawit-metabase reviewed (approved):
I have some outstanding doubts but approving based on the fact that the positive of test not flaking far outweighs the small negative of confusing logic.

jeff303 committed:
Ditch the "legacydriver_" prefix from the legacy BigQuery driver when creating new datasets; it's no longer needed now that transient datasets will have a much more unique suffix

Update the logic for the transient dataset suffix so that it can be more easily detected/parsed later (for cleanup)

Verbosify lots of comments 📜

jeff303 committed:
Fix bugs

jeff303 committed:
Ignore exceptions on failure to delete old transient datasets (other jobs may be doing the same)

jeff303 committed:
Remove unneeded if from `make-dataset-name-unique`

jeff303 committed:
Simplify `transient-dataset-outdated?` implementation

Add missing docstrings
